realm/realm-library/src/objectServer/java/io/realm/SyncConfiguration.java

```diff
@@ -257,6 +259,7 @@ boolean isSyncConfiguration() {
         private SharedRealm.Durability durability = SharedRealm.Durability.FULL;
         private boolean deleteRealmOnLogout = false;
         private final Pattern pattern = Pattern.compile("^[A-Za-z0-9_\\-\\.]+$"); // for checking serverUrl
+        private boolean readOnly = false;
```

Comment:
The pattern is final? I do agree that the mix of uninitialized and default values is rather ugly though.